web/src/main/java/org/cbioportal/web/parameter/StudyPageSettings.java

```diff
@@ -0,0 +1,256 @@
+package org.cbioportal.web.parameter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.validation.constraints.Size;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+/**
+ * @author kalletlak
+ *
+ */
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(Include.NON_NULL)
+public class StudyPageSettings extends PageSettingsData implements Serializable {
+    
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+    
+    @JsonInclude(Include.NON_NULL)
+    static class ChartSetting implements Serializable {
+
+        /**
+         * 
+         */
+        private static final long serialVersionUID = 1L;
+
+        static class Layout implements Serializable {
+
+            /**
+             * 
+             */
+            private static final long serialVersionUID = 1L;
+
+            private Integer x;
+            private Integer y;
+            private Integer w;
+            private Integer h;
+
+            public Integer getX() {
+                return x;
+            }
+
+            public void setX(Integer x) {
+                this.x = x;
+            }
+
+            public Integer getY() {
+                return y;
+            }
+
+            public void setY(Integer y) {
+                this.y = y;
+            }
+
+            public Integer getW() {
+                return w;
+            }
+
+            public void setW(Integer w) {
+                this.w = w;
+            }
+
+            public Integer getH() {
+                return h;
+            }
+
+            public void setH(Integer h) {
+                this.h = h;
+            }
+        }
+
+        @JsonInclude(Include.NON_NULL)
+        static class PatientSampleIdentifier extends SampleIdentifier implements Serializable {
+            /**
+             * 
+             */
+            private static final long serialVersionUID = 1L;
```

Comment:
What's this for?

Reply:
this is using in (de)serializing java object. probably autogenerated with it `implements Serializable`. I'll remove it as by default it generates one at runtime if not passed